### PR TITLE
Allow managing the required settigs for the email attribute

### LINKED ISSUE #15026: Declarative user profile should allow to mark the email attribute as non required
### Description

Following #8741, I was expecting "declarative user profile" would allow me to mark the user registration email field as non required.
Looks like it's not the case.

### Discussion

_No response_

### Motivation

Many of our business users don't have email addresses to provide.

### Details

_No response_

### PR DESCRIPTION
**Pull request by pedroigor**
Closes #15026

As a follow-up, we should be looking at the new admin UI as per https://github.com/keycloak/keycloak/issues/17974.

### CODE REVIEW

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/actions/AppInitiatedActionUpdateProfileTest.java [hunk 1]
```diff
@@ -258,7 +262,10 @@ public void updateProfileMissingEmail() {
         Assert.assertEquals("New last", updateProfilePage.getLastName());
         Assert.assertEquals("", updateProfilePage.getEmail());
 
-        Assert.assertEquals("Please specify email.", updateProfilePage.getInputErrors().getEmailError());
+        assertThat(updateProfilePage.getInputErrors().getEmailError(), anyOf(
```
Comment (vmuzikar) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/actions/AppInitiatedActionUpdateProfileTest.java:265]: Isn't `anyOf` a bit too nondeterministic for a test? :) But I'm probably just missing some context why we need to check for both error messages.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/actions/RequiredActionUpdateProfileTest.java [hunk 2]
```diff
@@ -226,7 +229,10 @@ public void updateProfileMissingEmail() {
         Assert.assertEquals("New last", updateProfilePage.getLastName());
         Assert.assertEquals("", updateProfilePage.getEmail());
 
-        Assert.assertEquals("Please specify email.", updateProfilePage.getInputErrors().getEmailError());
+        assertThat(updateProfilePage.getInputErrors().getEmailError(), anyOf(
```
Comment (vmuzikar) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/actions/RequiredActionUpdateProfileTest.java:232]: Same comment as above.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/forms/RegisterWithUserProfileTest.java [hunk 3]
```diff
@@ -602,6 +604,64 @@ public void testRegisterUserSuccess_attributeRequiredButNotSelectedByScopeIsNotR
...
+                + "{\"name\": \"email\"," + PERMISSIONS_ALL + ", \"required\": {}}"
+                + "]}");
+
+        loginPage.open();
+        loginPage.clickRegister();
+        registerPage.assertCurrent();
+
+        registerPage.register("firstName", "lastName", null, "registerWithoutEmail", "password", "password");
+        registerPage.assertCurrent();
+        assertThat(registerPage.getInputAccountErrors().getEmailError(), anyOf(
```
Comment (vmuzikar) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/forms/RegisterWithUserProfileTest.java:639]: Same as above.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/forms/RegisterWithUserProfileTest.java [hunk 4]
```diff
@@ -602,6 +604,64 @@ public void testRegisterUserSuccess_attributeRequiredButNotSelectedByScopeIsNotR
...
+                + "{\"name\": \"lastName\"," + PERMISSIONS_ALL + ", \"required\": {}},"
+                + "{\"name\": \"email\"," + PERMISSIONS_ALL + ", \"required\": {\"roles\" : [\"user\"]}}"
+                + "]}");
+
+        loginPage.open();
+        loginPage.clickRegister();
+        registerPage.assertCurrent();
+
+        registerPage.register("firstName", "lastName", null, "registerWithoutEmail", "password", "password");
+        assertThat(registerPage.getInputAccountErrors().getEmailError(), anyOf(
```
Comment (vmuzikar) [inline testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/forms/RegisterWithUserProfileTest.java:660]: Same as above.

**Review by vmuzikar — APPROVED** [review]
LGTM if the tests are green. Thanks @pedroigor!

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/actions/AppInitiatedActionUpdateProfileTest.java (hunk 1 above)
Comment (pedroigor) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/actions/AppInitiatedActionUpdateProfileTest.java:265]: Legacy and declarative user profile have different messages.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/actions/RequiredActionUpdateProfileTest.java (hunk 2 above)
Comment (pedroigor) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/actions/RequiredActionUpdateProfileTest.java:232]: Same as above.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/forms/RegisterWithUserProfileTest.java (hunk 3 above)
Comment (pedroigor) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/forms/RegisterWithUserProfileTest.java:639]: Same as above.

File: testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/forms/RegisterWithUserProfileTest.java (hunk 4 above)
Comment (pedroigor) [reply testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/forms/RegisterWithUserProfileTest.java:660]: Same as above.

**Pull request merged**